app-service-proxy/internal/server/transport.go

```diff
@@ -72,6 +72,24 @@ func (t *RewritingTransport) RoundTrip(req *http.Request) (*http.Response, error
 		return nil, err
 	}
 
+	// Patch CSP header
+	cspHeaders := resp.Header.Values("Content-Security-Policy")
+	if len(cspHeaders) > 0 {
+		var newCSPs []string
+		for _, csp := range cspHeaders {
+			if strings.Contains(csp, "frame-ancestors 'none'") {
```

Comment:
Looks good. I think there might be one other case worth looking at - if the `frame-ancestor` was set to a specific hostname, and did not have `self` then we would need to add in `self`.